src/Compilers/CSharp/Portable/Compiler/TypeCompilationState.cs

```diff
@@ -118,6 +118,7 @@ public bool Emitting
         public ArrayBuilder<MethodWithBody> SynthesizedMethods
         {
             get { return _synthesizedMethods; }
+            set { _synthesizedMethods = value; }
```

Comment:
This line worries me a bit.  It feels like it's opening us to the possibility of losing synthesized methods in the future as we can now replace vs. append.